src/coreclr/jit/valuenum.cpp

```diff
@@ -7813,7 +7813,195 @@ ValueNum ValueNumStore::EvalHWIntrinsicFunBinary(var_types      type,
+            simd64_t newVec = simd64_t::Zero();
+            CopyConstantSimd<simd64_t, TSimd, float>(&newVec, src);
+            return vns->VNForSimd64Con(newVec);
+        }
 #endif
+        case TYP_SIMD8:
+        default:
+        {
+            unreached();
+        }
+    }
+}
+
+ValueNum ExtendConstantSimdFloat(ValueNumStore* vns, var_types dstType, ValueNum srcVN)
+{
+    assert(vns->IsVNConstant(srcVN));
+
+    switch (vns->TypeOfVN(srcVN))
+    {
+        case TYP_SIMD8:
+        {
+            simd8_t src = vns->GetConstantSimd8(srcVN);
+            return ExtendConstantSimdFloat<simd8_t>(vns, dstType, &src);
+        }
+        case TYP_SIMD12:
+        {
+            simd12_t src = vns->GetConstantSimd12(srcVN);
+            return ExtendConstantSimdFloat<simd12_t>(vns, dstType, &src);
+        }
+        case TYP_SIMD16:
+        {
+            simd16_t src = vns->GetConstantSimd16(srcVN);
+            return ExtendConstantSimdFloat<simd16_t>(vns, dstType, &src);
+        }
+#if defined TARGET_XARCH
+        case TYP_SIMD32:
+        {
+            simd32_t src = vns->GetConstantSimd32(srcVN);
+            return ExtendConstantSimdFloat<simd32_t>(vns, dstType, &src);
+        }
+        case TYP_SIMD64:
+        {
+            simd64_t src = vns->GetConstantSimd64(srcVN);
+            return ExtendConstantSimdFloat<simd64_t>(vns, dstType, &src);
+        }
+#endif
+        default:
+        {
+            unreached();
+        }
+    }
+}
+
+ValueNum ValueNumStore::EvalHWIntrinsicFunTernary(var_types      type,
+                                                  var_types      baseType,
+                                                  NamedIntrinsic ni,
+                                                  VNFunc         func,
+                                                  ValueNum       arg0VN,
+                                                  ValueNum       arg1VN,
+                                                  ValueNum       arg2VN,
+                                                  bool           encodeResultType,
+                                                  ValueNum       resultTypeVN)
+{
+    if (IsVNConstant(arg0VN) && IsVNConstant(arg1VN) && IsVNConstant(arg2VN))
+    {
+
+        switch (ni)
+        {
+            case NI_Vector128_WithElement:
+#ifdef TARGET_ARM64
+            case NI_Vector64_WithElement:
+#else
+            case NI_Vector256_WithElement:
+            case NI_Vector512_WithElement:
+#endif
+            {
+                if (baseType != TYP_FLOAT) // No meaningful diffs for other base-types
+                {
+                    break;
+                }
+
+                var_types srcType = TypeOfVN(arg0VN);
+                if (srcType != type)
+                {
+                    if (genTypeSize(srcType) > genTypeSize(type))
+                    {
+                        break;
+                    }
+
+                    arg0VN = ExtendConstantSimdFloat(this, type, arg0VN);
```

Comment:
Not sure if this is worth the effort. It's only hit for Vector3/4 ctors which take a smaller vector plus some float(s) as arguments.

Reply:
Yes, I don't think it's worth it so please remove it